verdict/core/primitive.py

```diff
@@ -243,113 +244,141 @@ def populate_prompt_message(self, input: Schema, logger: Logger) -> PromptMessag
 
     def execute(
             self,
-            input: InputSchemaT
+            input: InputSchemaT,
+            tracer: Optional[Tracer] = None,
+            trace_id: Optional[str] = None,
```

Comment:
Can we wrap tracer/trace_id/parent_id in a Tracer/ExecutionContext dataclass (like MaterializationContext) and then recursively pass this down the execution chain?

Reply:
@qw3rtman , nice catch. I refactored the code to make this more standardized.